src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -123,113 +111,34 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
-            WDJ::JsonObject messageObject;
-
-            // _ActiveCommandline should be set already, we request for it the moment we become visible
-            winrt::hstring engineeredPrompt{ promptCopy + L". The shell I am running is " + _ActiveCommandline };
-            messageObject.Insert(L"role", WDJ::JsonValue::CreateStringValue(L"user"));
-            messageObject.Insert(L"content", WDJ::JsonValue::CreateStringValue(engineeredPrompt));
-            _jsonMessages.Append(messageObject);
-            jsonContent.SetNamedValue(L"messages", _jsonMessages);
-            jsonContent.SetNamedValue(L"max_tokens", WDJ::JsonValue::CreateNumberValue(800));
-            jsonContent.SetNamedValue(L"temperature", WDJ::JsonValue::CreateNumberValue(0.7));
-            jsonContent.SetNamedValue(L"frequency_penalty", WDJ::JsonValue::CreateNumberValue(0));
-            jsonContent.SetNamedValue(L"presence_penalty", WDJ::JsonValue::CreateNumberValue(0));
-            jsonContent.SetNamedValue(L"top_p", WDJ::JsonValue::CreateNumberValue(0.95));
-            jsonContent.SetNamedValue(L"stop", WDJ::JsonValue::CreateStringValue(L"None"));
-            const auto stringContent = jsonContent.ToString();
-            WWH::HttpStringContent requestContent{
-                stringContent,
-                WSS::UnicodeEncoding::Utf8,
-                L"application/json"
-            };
-
-            request.Content(requestContent);
-
-            // Send the request
-            try
-            {
-                const auto response = _httpClient.SendRequestAsync(request).get();
-                // Parse out the suggestion from the response
-                const auto string{ response.Content().ReadAsStringAsync().get() };
-                const auto jsonResult{ WDJ::JsonObject::Parse(string) };
-                if (jsonResult.HasKey(L"error"))
-                {
-                    const auto errorObject = jsonResult.GetNamedObject(L"error");
-                    result = errorObject.GetNamedString(L"message");
-                }
-                else
-                {
-                    if (_verifyModelIsValidHelper(jsonResult))
-                    {
-                        const auto choices = jsonResult.GetNamedArray(L"choices");
-                        const auto firstChoice = choices.GetAt(0).GetObject();
-                        const auto messageObject = firstChoice.GetNamedObject(L"message");
-                        result = messageObject.GetNamedString(L"content");
-                        isError = false;
-                    }
-                    else
-                    {
-                        result = RS_(L"InvalidModelMessage");
-                    }
-                }
-            }
-            catch (...)
-            {
-                result = RS_(L"UnknownErrorMessage");
-            }
+        // Switch back to the foreground thread because we are changing the UI now
+        co_await winrt::resume_foreground(Dispatcher());
```

Comment:
None of these member access including `Dispatcher()` are safe. You need to hold onto a "intelligent" this pointer or otherwise you won't know when `this` got destroyed.